packages/commonwealth/client/scripts/views/pages/CreateCommunity/steps/BasicInformationStep/BasicInformationForm/validation.ts

```diff
@@ -0,0 +1,29 @@
+import { VALIDATION_MESSAGES } from 'helpers/formValidationMessages';
+import z from 'zod';
+
+export const socialLinkValidation = z
+  .string()
+  .url({ message: VALIDATION_MESSAGES.INVALID_INPUT });
+
+export const basicInformationFormValidationSchema = z.object({
+  communityName: z
+    .string({ invalid_type_error: VALIDATION_MESSAGES.NO_INPUT })
+    .nonempty({ message: VALIDATION_MESSAGES.NO_INPUT })
+    .max(255, { message: VALIDATION_MESSAGES.MAX_CHAR_LIMIT_REACHED }),
+  communityDescription: z
+    .string({ invalid_type_error: VALIDATION_MESSAGES.NO_INPUT })
+    .nonempty({ message: VALIDATION_MESSAGES.NO_INPUT }),
+  communityProfileImageURL: z
+    .string({ invalid_type_error: VALIDATION_MESSAGES.NO_INPUT })
+    .nonempty({ message: VALIDATION_MESSAGES.NO_INPUT }),
+  chain: z.object(
+    {
+      value: z.any().default(-1).optional(),
+      label: z.string().default('').optional(),
+    },
+    {
+      invalid_type_error: VALIDATION_MESSAGES.NO_INPUT,
+      required_error: VALIDATION_MESSAGES.NO_INPUT,
+    },
+  ),
+});
```

Comment:
the idea was to use same schema for front end and backend validation so we should reuse the schema that was recently merged https://github.com/hicommonwealth/commonwealth/pull/5800

Reply:
I understand the approach, but the schema you are referring ([link](https://github.com/hicommonwealth/commonwealth/blob/451c19aab22b79cd20527d073dbd1b4016d68928/packages/commonwealth/shared/schemas/createCommunitySchema.ts#L30-L69)) won't be much beneficial here without a lot of emitting/updating since it has a lot of other fields unrelated to this form. If we go ahead and still use that with omitting keys then it can lead to snowballing complexity.

https://github.com/hicommonwealth/commonwealth/blob/451c19aab22b79cd20527d073dbd1b4016d68928/packages/commonwealth/shared/schemas/createCommunitySchema.ts#L30-L69

I am not sure what the next steps are for this one

cc: @jnaviask @masvelio

Reply:
I think that a majority of the schema is optional, so we won't need to worry about a majority of the fields.

I am not too sure since I am not super familiar with the zod integration with react hook form, but I think that we can choose specific fields to bind to the inputs, so I don't think it would add too much complexity.

But I will give it an attempt locally, and update here if it adds too much unnecessary complexity.

Reply:
Yeah so after playing around with it for a bit, I see the issue here. We are mixing our old validation with the react-hook-form validation inside CWTextInput. This makes it difficult to understand as well as add changes to. Not to mention our CWTextInput is ~300 lines long with 30 props. It is extremely fragile.

I see two options for next steps here depending on how urgent it is to get this in:
1. Refactor this out. Preferably we should not perform any react-form logic in our input components. This complicates the usage and mixes in this logic with our old error validation logic.
2. Just leave it for now and get this PR in using our old error handling ways.

Reply:
Just leaving some thoughts for 1

I actually think in the other direction. If we don't include new form validation logic in form control (input/radio etc) components then we would have to include them in form components, which in my opinion will lead to snowballing complexity and code redundancy.

What are your thoughts on this @masvelio

Reply:
Lmk what the next steps are for this one. 

If we want to update how form validation works then let's do that in another ticket, as this will get complex. For any nor-major actionable thing lmk and I will update the PR.


cc: @kurtisassad @masvelio

Reply:
Thanks @kurtisassad for bringing all of your concerns up. We do use `register` and other RHF utils but they are used in a bit different way (by using `useFormContext` hook). I see your points though, and I agree that we should have simple way to create form with both custom & backend validation. Important thing also would be to able to use shared zod schema, as we will go in this direction. 

Most important thing at this moment is to make create community form working properly, as we are tight on time and we need this PR in. So we can't afford refactor right now. Having said that, @mzparacha could you create a ticket, where we can explore how to adjust our components with react-hook-form, so that we can easily use shared schema.

Reply:
I am not exactly following the whole TRPC, shared zoc schema etc, but still created a blank ticket https://github.com/hicommonwealth/commonwealth/issues/5924. @masvelio / @kurtisassad can any of you plz add action items to the tickets as required. Thanks

Reply:
I agree. Any refactoring would be outside the scope of this PR. Feel free to resolve conflict after we make the spike ticket.